R/counts.R

```diff
@@ -180,6 +179,151 @@ get_single_cell_experiment <- function(
   sce
 }
 
+#' Gets a SummarizedExperiment from curated metadata
+#'
+#' Given a data frame of Curated Atlas metadata obtained from [get_metadata()],
+#' returns a [`SummarizedExperiment::SummarizedExperiment-class`] object
+#' corresponding to the samples in that data frame
+#'
+#' @param data A data frame containing, at minimum, a `sample_` column, which
+#'   corresponds to a single cell sample ID. This can be obtained from the
+#'   [get_metadata()] function.
+#' @param assays A character vector whose elements must be either "counts"
+#'   and/or "cpm", representing the corresponding assay(s) you want to request.
+#'   By default only the count assay is downloaded. If you are interested in
+#'   comparing a limited amount of genes, the "cpm" assay is more appropriate.
+#' @param repository A character vector of length one. If provided, it should be
+#'   an HTTP URL pointing to the location where the summarized experiment data is stored.
+#' @param cache_directory An optional character vector of length one. If
+#'   provided, it should indicate a local file path where any remotely accessed
+#'   files should be copied.
+#' @param features An optional character vector of features (ie genes) to return
+#'   the counts for. By default counts for all features will be returned.
+#' @returns A SummarizedExperiment object, with one assay for each value in the
+#'   assays argument
```

Comment:
A lot of this is a copy from `get_single_cell_experiment`. I would consider reading https://cran.r-project.org/web/packages/roxygen2/vignettes/reuse.html. Specifically, I can see `@inheritParams` being useful.